go/mysql/datetime/spec.go

```diff
@@ -374,10 +374,7 @@ func (u fmtWeek0) parse(t *timeparts, bytes string) (string, bool) {
 type fmtWeek1 struct{}
 
 func (fmtWeek1) format(dst []byte, t DateTime, prec uint8) []byte {
-	year, week := t.Date.ISOWeek()
-	if year < t.Date.Year() {
-		week = 0
-	}
+	week := t.Date.Week(1)
```

Comment:
I can make the same change for the other `fmtWeek` functions. Haven't done that yet, but i think it makes sense to just use the same implementation for everything.

Reply:
@GuptaManan100 i think we should do that indeed. Makes it less error prone and the week function afaik is already better tested too.

Reply:
@GuptaManan100 Pushed up the fix to always used the documented mode number.